### ISSUE TITLE
Improve triggering of ondemand update tests

### ISSUE THREAD

**Issue by whimboo** [mozilla/mozmill-ci#535]
As discussed  2 days ago we want to improve the triggering mechanics for the ondemand update tests. It should be easy enough and should not take someone to spend half an hour to get it configured and running. Therefore the following items have to be taking care off:

Parameters to use for the job:
* Version of target build to update to (build id will be automatically retrieved via FTP per platform)
* Earliest version of Firefox to use for updates (dropdown with default to oldest ESR)
* Update channel to use (free form field)

General behavior which applies to all kind of builds:
* Test en-US for every source build, and on every platform version
* Any locale should be covered only once per platform
* Ensure to run ja-JP-mac only on OS X, and ja on Windows and Linux
* If a locale is not present for the source build do not run the update test for it
* Oldest version of Firefox to use will default to the oldest supported ESR (train)

Specifics for Beta
* The former beta release we should always test even if it crosses the version boundary (e.g. 34.0b11 for the upcoming 35.b1 release)
* Otherwise pick a random build from every train (33, 32, 31, ...)
* (FUTURE) If there is a version boundary, pick latest RC instead of latest beta of former release

Specifics for Release
* Every release has to run on each of the OS versions by varying between remaining locales 

Specifics for ESR
* Oldest build to update from the current ESR branch is always the .0 release of that branch
* Major upgrade tests from an earlier ESR will have to be done via the trigger_ondemand script 

@KaiRo-at, @mwobensmith, and @kjozwiak can you please all have a look at this proposal? It would be good to finalize the specification soon. Thanks!

**Comment by KaiRo-at** [comment]
Sorry, this is from what I can tell identical to the notes from our meeting in Portland, so I was taken for granted that you know I agree with this and am looking forward to the implementation.

**Comment by ctalbert** [comment]
I told this to David Burns, but I'll note it here too. We need to identify exactly what we need from releng in order to make triggering these tests as simple as possible. We know that when releng automates uploading the snippets/binaries for activating releases, we will get a message. Let's ensure that message contains all the information we need so that we have to do the least amount of guessing about what it is we should test. Once we have that info, we should open a releng bug with exactly what we need. Many thanks.

**Comment by whimboo** [comment]
Clint, exactly that's why I want to have explicit feedback from the QA people on what I have written down. So if Matt and Kamil who manage the ESR releases could also have a look at I would really appreciate. I will put together an update notification listener shortly so we can see what we currently have. As @catlee mentioned to me during in Portland those notifications should exist, and additional properties can be added if needed. The only problem for the time being are water-shed releases with multi-step updates. If we cannot fix that on releng side, we might be able with the upcoming Marionette update tests, which will be able to support multiple updates in a row. CC#ing also @AutomatedTester so he is aware of this discussion.

**Comment by lizzard** [comment]
This will be really helpful. Henrik, the details of your proposal look accurate to what we've been doing for the last year.    

One thing that we don't have, but that would be very useful, would be to know (maybe with an email to mozmill-ci) that a batch of tests has started along with how many tests are in the batch, and another notification when the batch has finished.

**Comment by whimboo** [comment]
Thank you Liz for the feedback. It's good to see that all is covered. Regarding your proposal this is something very hard to implement because all the individual update jobs are running independently. But please file a new issue which exactly describes what you want to see.

**Comment by mwobensmith** [comment]
I'm fine with this. Thanks Henrik.

**Comment by whimboo** [comment]
We discussed this topic last week together with @bhearsum. As a final goal here we want to have our tests running on the RelEng infrastructure. This gives all of us more tighter control on when the tests have to be run. When that has been done no-one of you would have to trigger the tests via mozmill-ci!

So our plan is to finish the convertion of the update tests from Mozmill to Marionette and get them as stable as possible. That means we will get them running for Nightly builds on Mozmill-CI soon. Other versions of Firefox will still make use of Mozmill. Early in May we should have killed all regressions and should be able to get started on a transition. But keep in mind that the lowest version of Firefox we could run Marionette tests one will most likely be 38.0. 

Given the above I'm going to close this bug as wontfix. All the discussed requirements will be taken over to the list of things we need from releng. So those are not lost.

**Comment by whimboo** [comment]
As reference this is the [releng bug](https://bugzilla.mozilla.org/show_bug.cgi?id=813629)
